src/main/clojure/clara/rules/engine/helpers.cljc

```diff
@@ -0,0 +1,66 @@
+(ns clara.rules.engine.helpers
+  (:require
+    [clara.rules.engine.impl :as impl]
+    [clara.rules.engine.wme :as wme]))
+
+(defn has-keys?
+  "Returns true if the given map has all of the given keys."
+  [m keys]
+  (every? (partial contains? m) keys))
+
+(defn matches-some-facts?
+  "Returns true if the given token matches one or more of the given elements."
+  [token elements join-filter-fn condition]
+  (some (fn [{:keys [fact]}]
+          (join-filter-fn token fact (:env condition)))
+        elements))
+
+(defn send-accumulated
```

Comment:
This and other functions in this namespace are _only_ used in <snip>/engine/nodes/accumulators.cljc. This isn't so much of a reusable function as an implementation detail of the accumulator nodes. Therefore I think it might be best to simply keep these as private functions in that accumulators namespace so they remain a private implementation detail.